owncloudApp/src/androidTest/java/com/owncloud/android/sharing/shares/ui/TestShareFileActivity.kt

```diff
@@ -0,0 +1,67 @@
+/**
+ *   ownCloud Android client application
+ *
+ *   @author David González Verdugo
+ *   Copyright (C) 2019 ownCloud GmbH.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License version 2,
+ *   as published by the Free Software Foundation.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.owncloud.android.sharing.shares.ui
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.transaction
+import com.owncloud.android.R
+import com.owncloud.android.datamodel.OCFile
+import com.owncloud.android.domain.sharing.shares.model.OCShare
+import com.owncloud.android.presentation.ui.sharing.fragments.ShareFragmentListener
+import com.owncloud.android.testing.SingleFragmentActivity
+
+class TestShareFileActivity : SingleFragmentActivity(), ShareFragmentListener {
```

Comment:
It's the activity used with `ActivityScenario` and in which we set the fragments we want to test

Reply:
I don't find my previous comment. If you had a dependency with an activity, it would be great to have an interface ActivityProvider and be able to mock it. But now it is done in that way hehe

Reply:
Using FragmentScenario was not possible because of [this line ](https://github.com/owncloud/android/blob/dbc9e183b2a2f55d734eb72136b6f8dfb30c9cab/owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/ShareFileFragment.kt#L318) in `onAttach()` method. If it were possible to inject that context and use FragmentScenario at the same time it could work but I did not manage to do it.